File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/lint/LintProblem.java
package com.github.jmlparser.lint;

import com.github.javaparser.TokenRange;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author Alexander Weigl
 * @version 1 (13.10.22)
 */
public record LintProblem(
        @NotNull
        String level,
        @NotNull
        String message,
        @Nullable
        TokenRange location,
        @Nullable
        Throwable cause,
        @Nullable
        String category,
        @NotNull
        String ruleId) {

    public LintProblem(@NotNull String level, @NotNull String message, @Nullable TokenRange location, @NotNull String ruleId) {
        this(level, message, location, null, null, ruleId);
    }
}

File: tools/lint/src/main/kotlin/io/github/jmltoolkit/lint/LintProblem.kt
package io.github.jmltoolkit.lint

import com.github.javaparser.TokenRange

/**
 * @author Alexander Weigl
 * @version 1 (13.10.22)
 */
data class LintProblem(
    val level: String,
    val message: String,
    val location: TokenRange?,
    val cause: Throwable?,
    val category: String?,
    val ruleId: String
) {
    constructor(level: String, message: String, location: TokenRange?, ruleId: String) : this(
        level, message, location,
        null, null, ruleId
    )
}
